### PR TITLE
Fixed #24635 -- Updated deprecated urls.py examples in default project template.

### PR DESCRIPTION
**Pull request by jambonrose**
Changes according to https://code.djangoproject.com/ticket/24635

I've modified the `startproject` URL patterns examples to reflect deprecations in Django 1.8.

### CODE REVIEW

File: django/conf/project_template/project_name/urls.py [hunk 1]
```diff
@@ -2,9 +2,15 @@
 from django.contrib import admin
 
 urlpatterns = [
-    # Examples:
-    # url(r'^$', '{{ project_name }}.views.home', name='home'),
-    # url(r'^blog/', include('blog.urls')),
+    # Example 1: Function Views
```
Comment (timgraham) [inline django/conf/project_template/project_name/urls.py]: I think we could just say, "Function views:" and omit "Example X" for each case

File: django/conf/project_template/project_name/urls.py [hunk 2]
```diff
@@ -2,9 +2,15 @@
...
+    #   Step 1. Import view above:  from {{ project_name }}.views import home
```
Comment (timgraham) [inline django/conf/project_template/project_name/urls.py]: 1. Add this import: from {{ project_name }} import views

File: django/conf/project_template/project_name/urls.py [hunk 3]
```diff
@@ -2,9 +2,15 @@
...
+    #   Step 2. In this list:       url(r'^$', home, name='home'),
```
Comment (timgraham) [inline django/conf/project_template/project_name/urls.py]: 2. Add a URL to urlpatterns: url(r'^$', views.Home.as_view(), name='home'),

File: django/conf/project_template/project_name/urls.py [hunk 4]
```diff
@@ -2,9 +2,15 @@
...
+    # Example 2: Class-Based Views
+    #   Step 1. Import view above:  from {{ project_name }}.views import Home
+    #   Step 2. In this list:       url(r'^$', Home.as_view(), name='home'),
+    # Example 3: Include URL Configuration
+    #   Step 1. Import above:       from blog import urls as blog_urls
+    #   Step 2. In this list:       url(r'^blog/', include(blog_urls)),
```
Comment (timgraham) [inline django/conf/project_template/project_name/urls.py]: Dotted path isn't deprecated for `include()`

File: django/conf/project_template/project_name/urls.py [hunk 5]
```diff
@@ -2,9 +2,15 @@
 from django.contrib import admin
 
 urlpatterns = [
-    # Examples:
-    # url(r'^$', '{{ project_name }}.views.home', name='home'),
-    # url(r'^blog/', include('blog.urls')),
+    # Example 1: Function Views
+    #   Step 1. Import view above:  from {{ project_name }}.views import home
+    #   Step 2. In this list:       url(r'^$', home, name='home'),
+    # Example 2: Class-Based Views
```
Comment (timgraham) [inline django/conf/project_template/project_name/urls.py]: Class-based views

File: django/conf/project_template/project_name/urls.py [hunk 6]
```diff
@@ -2,9 +2,15 @@
...
+    #   Step 1. Import view above:  from {{ project_name }}.views import Home
+    #   Step 2. In this list:       url(r'^$', Home.as_view(), name='home'),
+    # Example 3: Include URL Configuration
```
Comment (timgraham) [inline django/conf/project_template/project_name/urls.py]: 3. Including another URLconf

**Comment by timgraham** [comment]
Could you add a note to the 1.8.1 release notes as well? Thanks!

File: django/conf/project_template/project_name/urls.py (hunk 1 above)
Comment (jambonrose) [inline django/conf/project_template/project_name/urls.py]: I would like to have the word 'Example' or examples somewhere, as I think that this makes it as clear as possible for beginners.

File: django/conf/project_template/project_name/urls.py (hunk 4 above)
Comment (jambonrose) [inline django/conf/project_template/project_name/urls.py]: True. However, I was under the impression that avoiding the namespace string was considered best practice, and opted to provide that instead. Am I mistaken?

**Comment by timgraham** [comment]
What do you think about using a module docstring since this has gotten more complicated?
```
"""
Examples:
1. Function views
  a. Add an import:  from {{ project_name }} import views
  b. Add a URL to urlpatterns:  url(r'^$', views.home, name='home')
2. Class-based views
  a. Add an import:  from {{ project_name }}.views import Home
  b. Add a URL to urlpatterns:  url(r'^$', Home.as_view(), name='home')
3. Including another URLconf
  a. Add an import:  from blog import urls as blog_urls
  b. Add a URL to urlpatterns:  url(r'^blog/', include(blog_urls))
"""
```

File: docs/releases/1.8.1.txt [hunk 7]
```diff
@@ -38,3 +38,7 @@ Bugfixes
 
 * Prevented arbitrary file inclusions in :mod:`~django.contrib.admindocs`
   (:ticket:`24625`).
+
+* Updated ``urlpatterns`` examples generated by :djadmin:`startproject` to
```
Comment (timgraham) [inline docs/releases/1.8.1.txt]:
```
* Updated ``urlpatterns`` examples generated by :djadmin:`startproject` to
  remove usage of referencing views by dotted path in :func:`django.conf.urls.url`
  which is deprecated in Django 1.8 (:ticket:`24635`).
```

**Comment by jambonrose** [comment]
I rather like the docstring syntax - but Python treats the docstring as a string in the list at that point, so we need to use standard comments.

I'm currently gathering feedback at the PyCon sprints. Will have a new version in a bit.

**Comment by timgraham** [comment]
Right, I was thinking to put the docstring at the top of the file.

**Comment by jambonrose** [comment]
Oh, right. That makes more sense. Given the current changes, I'm happy either way (inline or module docstring).

I took a poll about whether to prefix the examples and steps with "Example #:" and "Step", and found that people preferred the version without the prefixes (12 to 8).

I had a request to add a link to the documentation. People also pointed out that imports from the project didn't make a whole lot of sense, so I've made changes accordingly.

How's this look?

Thanks for all the feedback.

**Comment by jambonrose** [comment]
If we want to use a module docstring instead, I think the small changes below would do the trick:
```python
"""
The `urlpatterns` list routes URLs to views. For more information please see:
    https://docs.djangoproject.com/en/{{ docs_version }}/topics/http/urls/
Examples:
Function views
    1. Add this import:  from my_app import views
    2. Add a URL to urlpatterns:  url(r'^$', views.home, name='home'),
Class-based views
    1. Add this import:  from other_app.views import Home
    2. Add a URL to urlpatterns:  url(r'^$', Home.as_view(), name='home'),
Including another URLconf
    1. Add this import:  from blog import urls as blog_urls
    2. Add a URL to urlpatterns:  url(r'^blog/', include(blog_urls)),
"""
```

**Comment by jambonrose** [comment]
A quick poll shows people prefer the module docstring. Switching to that.

**Comment by timgraham** [comment]
merged in 0b5abb03218ff0c575cee0c83e446bab81cbfadd, thanks!

**Pull request closed without merging**